tests/test_generators/test_sqltablegen.py

```diff
@@ -137,6 +137,50 @@ def test_abstract_class(capsys):
     assert 'CREATE TABLE "inherited class"' in ddl2
 
 
+def test_index_sqlddl(capsys):
+    b = SchemaBuilder()
+    b.add_slot(SlotDefinition("age", range="integer", description="age of person in years"))
+    b.add_slot(SlotDefinition("dummy_foreign_key", range="ClassWithNowt", description="foreign key test"))
+    b.add_slot("identifier_slot", identifier=True)
+    b.add_slot("key_slot", key=True)
+    slots = ["full name", "description", "dummy_foreign_key", "age"]
+    # Simple Multicolumn index defined in annotation
+    test_index = Annotation(tag="index", value={"index2": ["id", "age"]})
+    # Duplicate Index Name
+    test_index_2 = Annotation(tag="index", value={"ix_ClassWithId_identifier_slot": ["identifier_slot", "name"]})
+    test_index_3 = Annotation(tag="index", value={"ClassWithNowt_slot_1_slot_2_idx": ["slot_1"]})
+    test_index_dict = {"index": test_index}
+    test_index_dict_2 = {"index": test_index_2}
+    test_index_dict_3 = {"index": test_index_3}
+    # testing to ensure
+    b.add_class(DUMMY_CLASS, slots, description="My dummy class", annotations=test_index_dict)
+    # testing to ensure the duplicated index isn't generated
+    b.add_class("ClassWithId", slots=["identifier_slot", "name", "whatever"], annotations=test_index_dict_2)
+    # Testing Unique Constraint
+    slot_1_2_UK = UniqueKey(unique_key_name="unique_keys", unique_key_slots=["slot_1", "slot_2"])
+    b.add_class(
+        "ClassWithNowt",
+        slots=["slot_1", "slot_2"],
+        annotations=test_index_dict_3,
+        unique_keys={"unique_keys": slot_1_2_UK},
+    )
+    gen = SQLTableGenerator(b.schema, use_foreign_keys=True)
+    ddl = gen.generate_ddl()
+    # Tests autogeneration of primary key index
+    assert 'CREATE INDEX "ix_dummy class_id" ON "dummy class" (id);' in ddl
```

Comment:
does this statement play nicely with the generator's rules about how to format the class name?

Reply:
Can you add tests where a naming policy is specified?

Reply:
Let's make the separate issue a high priority (maybe we can get it fixed before the next linkml release?).  We can also refine the syntax style in subsequent PRs.